src/OrbitGl/App.h

```diff
@@ -361,6 +361,20 @@ class OrbitApp final : public DataViewFactory, public CaptureListener {
   void SetCollectThreadStates(bool collect_thread_states);
   void SetMaxLocalMarkerDepthPerCommandBuffer(uint64_t max_local_marker_depth_per_command_buffer);
 
+  void SetCollectMemoryInfo(bool collect_memory_info) {
+    data_manager_->set_collect_memory_info(collect_memory_info);
+  }
+  [[nodiscard]] bool GetCollectMemoryInfo() const { return data_manager_->collect_memory_info(); }
+  void SetMemorySamplingPeriodNs(uint64_t memory_sampling_period_ns) {
```

Comment:
Maybe stupid question: We don't have GetMemorySamplingPeriod because we are not using it?

Reply:
Yep. This is not stupid at or :-)! For `memory_sampling_period_ns`, we will set it to `grpc_protos::CaptureOptions`. And when we need the value in the Service side, we can use `capture_option.memory_sampling_period_ns()`. 
`warning_threshold` is not a part of `grpc_protos::CaptureOptions`, so we have a Get method for it.